asset/asset.go

```diff
@@ -1098,15 +1145,55 @@ func (s *ScriptKey) HasScriptPath() bool {
 	return s.TweakedScriptKey != nil && len(s.TweakedScriptKey.Tweak) > 0
 }
 
+// DetermineType attempts to determine the type of the script key based on the
+// information available. This method will only return ScriptKeyUnknown if the
+// following condition is met:
+//   - The script key doesn't have a script path, but the final Taproot output
+//     key doesn't match a BIP-0086 key derived from the internal key. This will
+//     be the case for "foreign" script keys we import from proofs, where we set
+//     the internal key to the same key as the tweaked script key (because we
+//     don't know the internal key, as it's not part of the proof encoding).
+func (s *ScriptKey) DetermineType() ScriptKeyType {
+	// If we have an explicit script key type set, we can return that.
+	if s.TweakedScriptKey != nil &&
+		s.TweakedScriptKey.Type != ScriptKeyUnknown {
+
+		return s.TweakedScriptKey.Type
+	}
+
+	// If there is a known tweak, then we know that this is a script path
+	// key. We never return the channel type, since those keys should always
+	// be declared properly, and we never should need to guess their type.
+	if s.HasScriptPath() {
```

Comment:
Seems that for tap channel keys, we'd miss out on a more precise label here. Unless we always take a code path that skips this routine.

Reply:
Yes, that's correct. For already existing channel keys (excluding the funding `OP_TRUE` key, as that's detectable), we couldn't assign the correct value.
But those should only include force close outputs. And they should be spent already, so hopefully not causing too much confusion for users.